web/client/epics/geostory.js

```diff
@@ -114,7 +115,12 @@ const updateMediaSection = (store, path) => action$ =>
             let resourceId = resource.id;
             if (resourceAlreadyPresent) {
                 resourceId = resourceAlreadyPresent.id;
-            } else {
+            } else if (isEmpty(resource)) {
```

Comment:
this piece of code should be tested or see if it is covered by current tests.

you can check this by looking at the coverage report in coverage/index.html